org-roam.el

```diff
@@ -375,9 +375,41 @@ current buffer is used."
             (org-roam--format-title title file-path))
           (org-roam--extract-titles)))
 
+(defun org-roam--ref-type-p (type)
+  "Return t if the ref from current buffer is TYPE."
+  (let ((current (car (org-roam--extract-ref))))
+    (eq current type)))
+
 (defun org-roam--extract-ref ()
-  "Extract the ref from current buffer."
-  (cdr (assoc "ROAM_KEY" (org-roam--extract-global-props '("ROAM_KEY")))))
+  "Extract the ref from current buffer and return the type and the key of the ref."
+  (if-let ((ref (cdr (assoc "ROAM_KEY" (org-roam--extract-global-props '("ROAM_KEY"))))))
+      (let* ((type (org-roam--ref-type ref))
+             (key (cond ((string= "cite" type)
+                         (s-chop-prefix (org-roam--cite-prefix ref) ref))
+                        (t ref))))
+        (cons type key))
+    nil))
+
+(defun org-roam--ref-type (ref)
+  "Determine the type of the REF from the prefix."
+  (let* ((cite-prefix (org-roam--cite-prefix ref))
+         (is-website (seq-some
+                      (lambda (prefix) (s-prefix? prefix ref))
+                      '("http" "https")))
+         (type (cond (cite-prefix "cite")
+                     (is-website "website")
+                     (t "roam"))))
+    type))
+
+(defun org-roam--cite-prefix (ref)
+  "Return the citation prefix of REF (e.g. \"cite:\",\"parencite:\" etc.) if it has a prefix of
```

Comment:
The docstring needs to be tweaked to suit checkdoc requirements.

1. There's a faux indentation somewhere on this line.
2. The docstring needs start with a full sentence and REF must appear somwhere in it. Descriptions can come in later sentences.